### PR TITLE
[CARBONDATA-1283] Carbon should continue with the default value if wrong value is configured for the property

### PR DESCRIPTION
**Pull request by mohammadshahidkhan**

### CODE REVIEW

**Comment by asfgit** [comment]
Can one of the admins verify this patch?

**Comment by asfgit** [comment]
Can one of the admins verify this patch?

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/2993/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6, Please check CI http://144.76.159.231:8080/job/ApacheCarbonPRBuilder/404/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/3020/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 1.6, Please check CI http://144.76.159.231:8080/job/ApacheCarbonPRBuilder/432/

File: core/src/main/java/org/apache/carbondata/core/util/CarbonProperties.java [hunk 1]
```diff
@@ -98,6 +100,119 @@ private void validateAndLoadDefaultProperties() {
...
+  }
+
+  /**
+   * the method decide and set the lock type based on the configured system type
+   *
+   * @param lockTypeConfigured
+   */
+  private void configureDefaultLockType(String lockTypeConfigured) {
+    Configuration configuration = new Configuration(true);
+    String defaultFs = configuration.get("fs.defaultFS");
```
Comment (gvramana) [inline core/src/main/java/org/apache/carbondata/core/util/CarbonProperties.java]: How about the case when default.fs is hdfs but carbon store path is configured as local file path.
Alluxio fs what is behaviour of locking?

File: core/src/main/java/org/apache/carbondata/core/constants/CarbonCommonConstants.java [hunk 2]
```diff
@@ -329,6 +329,15 @@
    */
   public static final String CSV_READ_BUFFER_SIZE_DEFAULT = "50000";
   /**
+   * min value for csv read buffer size
+   */
+  public static final int CSV_READ_BUFFER_SIZE_MIN = 10240; //10 kb
+  /**
+   * max value for csv read buffer size
+   */
+  public static final int CSV_READ_BUFFER_SIZE_MAX = 10485760; // 10 mb
```
Comment (gvramana) [inline core/src/main/java/org/apache/carbondata/core/constants/CarbonCommonConstants.java]: This size can be 100mb.

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6, Please check CI http://144.76.159.231:8080/job/ApacheCarbonPRBuilder/447/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/3035/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6, Please check CI http://144.76.159.231:8080/job/ApacheCarbonPRBuilder/448/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/3036/

**Comment by gvramana** [comment]
LGTM

**Pull request closed without merging**